File: ecommerce-api-legacy/src/middlewares/errorHandler.js
class ApiError extends Error {
    constructor(message, status = 400) {
        super(message);
        this.status = status;
    }
}

// Middleware de erro centralizado do Express (registrado por último).
// Erros de domínio viram o status/mensagem apropriados; erros inesperados
// viram 500 sem vazar internals para o cliente.
function errorHandler(err, req, res, next) { // eslint-disable-line no-unused-vars
    if (err instanceof ApiError) {
        return res.status(err.status).send(err.message);
    }
    console.error('[ERROR]', err);
    return res.status(500).send('Erro interno');
}

module.exports = { ApiError, errorHandler };

File: task-manager-api/middlewares/error_handler.py
"""Error handling centralizado: erros de domínio viram status HTTP; inesperados viram 500 sem vazar internals."""
from flask import jsonify


class ApiError(Exception):
    """Erro de domínio com status HTTP associado."""

    def __init__(self, message, status=400):
        super().__init__(message)
        self.message = message
        self.status = status


def register_error_handlers(app):
    @app.errorhandler(ApiError)
    def handle_api_error(error):
        return jsonify({'error': error.message}), error.status

    @app.errorhandler(404)
    def handle_not_found(error):
        return jsonify({'error': 'Recurso não encontrado'}), 404

    @app.errorhandler(Exception)
    def handle_unexpected(error):
        app.logger.exception('Erro não tratado')
        return jsonify({'error': 'Erro interno'}), 500
